### ISSUE TITLE
WinRM provisioning timeout after 1 hour

### ISSUE THREAD

**Issue by nscaife** [mitchellh/vagrant#5505]
I believe this is the same issue as #4142.  Provisioning fails exactly one hour after the provisioning begins.  I'm trying to provision Visual Studio 2013, which takes almost 3 hours to install completely.  The specific error provided in the debug log is:

```
ERROR warden: Error occurred: An error occurred executing a remote WinRM command.

Shell: powershell
Command: powershell -ExecutionPolicy Bypass -OutputFormat Text -file c:\tmp\vagrant-shell.ps1
if ($?) { exit 0 } else { if($LASTEXITCODE) { exit $LASTEXITCODE } else { exit 1 } }
Message: execution expired
 INFO warden: Beginning recovery process...
 INFO warden: Recovery complete.
ERROR warden: Error occurred: An error occurred executing a remote WinRM command.
```

Here is the Vagrantfile.  I am using `privileged: false` on the `cinst` because, despite what the documentation says, it does change the behavior of the provisioner.  Without it, Vagrant installs a scheduled task which runs `elevated_shell.ps1` that actually runs my script.  This is a separate issue, but the created scheduled task has a fixed timeout of 2 hours.

```ruby
# -*- mode: ruby -*-
# vi: set ft=ruby :

Vagrant.require_version ">= 1.6.2"

Vagrant.configure("2") do |config|

    config.vm.define "dev", primary: true do |dev|
        dev.vm.define "dev"
        dev.vm.box = "eval-win81x64-enterprise-nocm"
        dev.vm.communicator = "winrm"
        dev.vm.host_name = "dev"
        dev.vm.guest = :windows
        dev.windows.halt_timeout = 15
        dev.vm.provision "shell", path: "vagrant-scripts/install_chocolatey.ps1"
        dev.vm.provision "shell", inline: "cinst -y visualstudiocommunity2013", privileged: false
    end

    # Admin user name and password
    config.winrm.username = "vagrant"
    config.winrm.password = "vagrant"

    config.winrm.timeout = 21600 # set timeout exorbitantly high for visual studio install

    config.vm.network :forwarded_port, guest: 3389, host: 3389, id: "rdp", auto_correct: true
    config.vm.network :forwarded_port, guest: 22, host: 2222, id: "ssh", auto_correct: true

    config.vm.boot_timeout = 900

    config.vm.provider :virtualbox do |v, override|
        v.gui = false
        v.customize ["modifyvm", :id, "--memory", 1024]
        v.customize ["modifyvm", :id, "--cpus", 2]
        v.customize ["setextradata", "global", "GUI/SuppressMessages", "all" ]
    end

end
```

On the guest, `winrm/config Maxtimeoutms` is set to `21600000` (6 hours).

Like @matt-richardson in #4142, I believe this may be related to a WinRM bug that sounds like it was corrected with WinRb/WinRM#101 (merged in version 1.3).  However, the embedded version in Vagrant is 1.1.3.

**Comment by todadot** [comment]
Hi, is there any info about this? I bumped into the same problem, and I would need long provisioning :S

**Comment by sethvargo** [comment]
Hi there,

Is this still an issue with Vagrant 1.7.3? Thanks!

**Comment by todadot** [comment]
When I was working with this there was no fix for vagrant. If I remember correctly what vagrant use for winrm it has an update for that but that is not integrated. You can manually modify the script which it does, but I did not pick that path, rahter I used PSTools at the end. (I do not remember where you have to modify the script but that is possible)

**Comment by mitchellh** [comment]
This will be fixed in 1.8. We now have a setting for setting the timeout.

**Comment by nscaife** [comment]
@mitchellh Just to confirm, this is a different issue than #5506, which was corrected in PR #6213.  This issue refers to an *additional* fixed timeout inside the WinRb component.
